src/django_components/slots.py

```diff
@@ -122,6 +127,63 @@ def __str__(self) -> str:
         return mark_safe(self._slot.nodelist.render(self._context))
 
 
+class SlotIsFilled(dict):
+    """
+    Dictionary that returns `True` if the slot is filled (key is found), `False` otherwise.
+    """
+
+    def __init__(self, fills: Dict, *args: Any, **kwargs: Any) -> None:
+        escaped_fill_names = {_escape_slot_name(fill_name): True for fill_name in fills.keys()}
+        super().__init__(escaped_fill_names, *args, **kwargs)
+
+    def __missing__(self, key: Any) -> bool:
+        return False
+
+
+@dataclass
+class ComponentSlotContext:
+    component_name: str
+    template_name: str
+    is_dynamic_component: bool
+    default_slot: Optional[str]
+    fills: Dict[SlotName, Slot]
+
+    def post_render_validation(self) -> None:
```

Comment:
This method runs after the component's template is done rendering. It's only at this point that we have the knowledge of all slots that were rendered, and we can potentially do more validation based on that.